### ISSUE TITLE
Integer ! and ~ operators are incorrect

### ISSUE THREAD

**Issue by ajor** [iovisor/bpftrace#647]
C:
```
#include <stdio.h>

int main()
{
  printf("%d %d %d\n", -10, !10, ~10);
  return 0;
}
```
```
-10 0 -11
```

bpftrace:
```
BEGIN
{
  printf("%d %d %d\n", -10, !10, ~10);
}
```
```
Attaching 1 probe...
-10 -11 -10
^C
```

**Comment by synth0** [comment]
Hey @ajor, would you mind if I worked on this?

I think I have an idea on how to go about this.

**Comment by ajor** [comment]
Go for it!

**Comment by synth0** [comment]
@ajor should this be closed now that #700 is merged?

**Comment by ajor** [comment]
Yeah, thanks
